### PR TITLE
Fix fastrand bug when p1 = 1.0.  int was overflowing.

### PR DESCRIPTION
**Pull request by andychu**
Add the ability to run an individual Python test with the right
PYTHONPATH.

### CODE REVIEW

File: client/python/_fastrand.c [hunk 1]
```diff
@@ -36,10 +36,12 @@ limitations under the License.
 
 uint64_t randbits(float p1, int num_bits) {
   uint64_t result = 0;
-  int threshold = (int)(p1 * RAND_MAX);
+  uint64_t threshold = p1 * RAND_MAX;
   int i;
   for (i = 0; i < num_bits; ++i) {
-    uint64_t bit = (rand() < threshold);
+    // NOTE: The comparison is <= so that p1 = 1.0 implies that the bit is
```
Comment (tkaitchuck) [inline client/python/_fastrand.c]: This comment does not reflect what the change to the code is doing, and both are wrong.
p=1.0 implies that the value is always zero. I think you meant to type p=0 implies the bit is always one. That would be true given the change you made. However it is undesirable. We don't want it to be the case that if someone sets p=0 they get a bit array of all 1s. It should be zeros. Less than is correct.

Comment (andychu) [inline client/python/_fastrand.c]: Not sure what you're saying here.  Did you look at the tests?  You get all
0's if p = 0.0, and all 1's if p = 1.0.

If the comparison were < rather than <=, then you would have an edge case
when rand() returns RAND_MAX.  RAND_MAX < RAND_MAX would fail and you would
get 0 instead of 1.  The idea is to have it all be 1's.

Andy

Comment (tkaitchuck) [inline client/python/_fastrand.c]: It clearly is not ALL ones. There should be no way for this function to
return all ones. If it can it is a bug.

**Comment by andychu** [comment]
Oh OK we're thinking about the opposite cases... < creates the edge case with RAND_MAX, and <= creats it with 0.  Maybe I can use RAND_MAX+1.  Let me look it up.

**Comment by tkaitchuck** [comment]
Consider there are three relevant cases p=0, p=1, 0<p<1
in the p=0 we need the result to be zero.
in the p=1 we need the result to be one.
in the other cases either way as it will be accurate to within 1/Max_rand

I think the solution is to subtract one from the generated value, as adding one to Max_rand may overflow.

**Comment by tkaitchuck** [comment]
Never mind. Max_rand has to be an integer as long as threshold is a uint it can hold max_rand+1

**Comment by andychu** [comment]
OK, I fixed it and added a big comment explaining it.
